builtin/logical/pki/path_root.go

```diff
@@ -236,6 +236,23 @@ func (b *backend) pathCAGenerateRoot(ctx context.Context, req *logical.Request,
 	resp.Data["key_id"] = myKey.ID
 	resp.Data["key_name"] = myKey.Name
 
+	// Update the issuer to reflect the PSS status here for revocation; this
+	// allows CRL building to succeed if the root is using a managed key with
+	// only PSS support.
+	if input.role.KeyType == "rsa" && input.role.UsePSS {
+		myIssuer.RevocationSigAlg = x509.SHA256WithRSAPSS
```

Comment:
I'm assuming there was already validation on setting role.SignatureBits so that some odd value doesn't just become SHA256 because it was set wrong?

Reply:
This was done after the key and issuer was already created, so SignatureBits is already used and validated earlier. 

See https://github.com/hashicorp/vault/blob/3436a7476f77b0f0c7965fe42dcf2b16472c6cc3/builtin/logical/pki/ca_util.go#L73 via https://github.com/hashicorp/vault/blob/3436a7476f77b0f0c7965fe42dcf2b16472c6cc3/builtin/logical/pki/ca_util.go#L20 via https://github.com/hashicorp/vault/blob/3436a7476f77b0f0c7965fe42dcf2b16472c6cc3/builtin/logical/pki/path_root.go#L114 at the top of the function. 

As far as safe values goes, 256 is still decent.

Reply:
Better variant with discussion in the code has now been pushed.